server/models/service-process.js

```diff
@@ -128,6 +128,23 @@ module.exports = function extendServiceProcess(ServiceProcess) {
   }
   ServiceProcess.recordProfilingState = recordProfilingState;
 
+  function recordStatusWdUpdate(instanceId, pInfo, callback) {
+    function updateProcessStatus(proc, asyncCb) {
+      proc.isTracing = pInfo.isTracing || false;
+      proc.save(asyncCb);
+    }
+
+    return async.waterfall([
+      _findProcess(instanceId, +pInfo.id, pInfo.pid, +pInfo.pst),
```

Comment:
I find function definition before the async hard to read in comparison to presenting code in the order it is executed.